[inductor] split out triton templates

coconutruben opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #156151
* #156273
* #156275
* #156150
* #155775
* #155774
* __->__ #155773

Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

coconutruben committed:
[inductor] split out triton templates

Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor] split out triton templates"


Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

[ghstack-poisoned]

masnesral reviewed (approved):
LGTM; looks like you have a few test failures that need a restart.

NikhilAPatel reviewed (approved):
LGTM

coconutruben committed:
Update on "[inductor] split out triton templates"


Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor] split out triton templates"


Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor] split out triton templates"


Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor] split out triton templates"


Summary:
# Why

Part of the work towards an interface for performance modeling and lookup table support

The goal is to have a clean separation of Templates, Params for those templates, and Inputs, and only rely on those to generate a ChoiceCaller (but also to reduce the search space for configs). This is a refactor towards that that enables cleaner subclassing and maintenance of the triton templates

# What

- move triton mm and tma templates out of mm.py and into triton_templates/ necessary bridges for that to work

# Context

These diffs are about making this work for mm and Triton (or rather, things inside mm.py) initially and then expanding out after those are landed to cover the other templates (bmm, conv, b2b_gemm, etc)

Test Plan:

```
buck2 run mode/opt scripts/coconutruben/torchmm:experiment 2>&1
buck2 run mode/opt-amd-gpu scripts/coconutruben/torchmm:experiment 2>&1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
